testkit/src/main/scala/org/bitcoins/rpc/BitcoindRpcTestUtil.scala

```diff
@@ -213,62 +243,193 @@ trait BitcoindRpcTestUtil extends BitcoinSLogger {
+              }
+            }
+          }
         }
-        (client1, client2)
       }
     }
   }
 
+  def signRawTransaction(
+      signer: BitcoindRpcClient,
+      transaction: Transaction,
+      utxoDeps: Vector[RpcOpts.SignRawTransactionOutputParameter] = Vector.empty,
+  )(implicit actorMaterializer: ActorMaterializer): Future[
+    SignRawTransactionResult] = signer match {
+    case v17: BitcoindV17RpcClient =>
+      v17.signRawTransactionWithWallet(transaction, utxoDeps)
+    case v16: BitcoindV16RpcClient =>
+      v16.signRawTransaction(transaction, utxoDeps)
+    case unknown: BitcoindRpcClient =>
+      val v16T = BitcoindV16RpcClient.fromUnknownVersion(unknown)
+      val v17T = BitcoindV17RpcClient.fromUnknownVersion(unknown)
+      (v16T, v17T) match {
+        case (Failure(_), Failure(_)) =>
+          throw new RuntimeException(
+            "Could not figure out version of provided bitcoind RPC client!")
+        case (Success(_), Success(_)) =>
+          throw new RuntimeException(
+            "This should not happen, managed to construct different versioned RPC clients from one single client")
+        case (Success(v16), Failure(_)) =>
+          v16.signRawTransaction(transaction, utxoDeps)
+        case (Failure(_), Success(v17)) =>
+          v17.signRawTransactionWithWallet(transaction, utxoDeps)
+      }
+  }
+
+  def sendCoinbaseTransaction(
+      sender: BitcoindRpcClient,
+      receiver: BitcoindRpcClient,
+      amount: Bitcoins = Bitcoins(1))(
+      implicit executionContext: ExecutionContext,
+      actorMaterializer: ActorMaterializer): Future[GetTransactionResult] = {
+    createRawCoinbaseTransaction(sender, receiver, amount)
+      .flatMap(signRawTransaction(sender, _))
+      .flatMap { signedTransaction =>
+        sender
+          .generate(100)
+          .flatMap { _ => // Can't spend coinbase until depth 100
+            sender
+              .sendRawTransaction(signedTransaction.hex, allowHighFees = true)
+              .flatMap { transactionHash =>
+                sender.getTransaction(transactionHash)
+              }
+          }
+      }
+  }
+
+  def getFirstBlock(
+      implicit
+      node: BitcoindRpcClient,
+      executionContext: ExecutionContext): Future[
+    GetBlockWithTransactionsResult] = {
+    node.getBlockHash(1).flatMap { hash =>
+      node.getBlockWithTransactions(hash)
+    }
+  }
+
+  def fundBlockChainTransaction(
+      sender: BitcoindRpcClient,
+      address: BitcoinAddress,
+      amount: Bitcoins)(
+      implicit ec: ExecutionContext): Future[DoubleSha256Digest] = {
+    fundMemPoolTransaction(sender, address, amount).flatMap { txid =>
+      sender.generate(1).map { _ =>
+        txid
+      }
+    }
+  }
+
+  def fundMemPoolTransaction(
+      sender: BitcoindRpcClient,
+      address: BitcoinAddress,
+      amount: Bitcoins)(
+      implicit ec: ExecutionContext): Future[DoubleSha256Digest] = {
+    sender
+      .sendToAddress(address, amount)
```

Comment:
This method was meant to be in the second commit, because it is a hack which shouldn't be there. I was having issues with transactions not being propagated/accepted into the mempool by doing just `sendToAddress`, so I had to go this weird route of `sendToAddress` -> `getTransaction` -> `sendRawTransaction`

Reply:
That is strange. Add this comment as a scaladoc on this method

Reply:
I was thinking I should try and remove it first, and see if the same errors are still there